### PR TITLE
Forbid usage of property "name" in Effect.Tag

### LINKED ISSUE #3286: Effect.Tag with non Effect-returning functions
### What version of Effect is running?

latest

### What steps can reproduce the bug?

https://effect.website/play#d5c06311f98e

### What is the expected behavior?

_No response_

### What do you see instead?

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by mikearnaldi**
closes #3286

### CODE REVIEW

File: packages/effect/src/Effect.ts [hunk 1]
```diff
@@ -5984,9 +5984,10 @@ export const Tag: <const Id extends string>(id: Id) => <
   & Context.TagClass<Self, Id, Type>
   & (Type extends Record<PropertyKey, any> ? {
       [
-        k in keyof Type as Type[k] extends ((...args: [...infer Args]) => infer Ret) ?
-          ((...args: Readonly<Args>) => Ret) extends Type[k] ? k : never
-          : k
+        k
+          in keyof Type as (Type[k] extends ((...args: [...infer Args]) => infer Ret) ?
+            ((...args: Readonly<Args>) => Ret) extends Type[k] ? k : never
+            : k) extends infer z extends PropertyKey ? z extends "name" ? never : z : never
```
Comment (tim-smart) [inline packages/effect/src/Effect.ts:5990]: Can we add it to `ProhibitedType`?

Comment (tim-smart) [reply packages/effect/src/Effect.ts:5990]: Or maybe we make the others "passthrough" like `name`.

Comment (mikearnaldi) [reply packages/effect/src/Effect.ts:5990]: what do you mean by passthrough?

Comment (mikearnaldi) [reply packages/effect/src/Effect.ts]: ah understood, no I think it is better to prohibit, less risk for confusion

File: packages/effect/src/Effect.ts [hunk 2]
```diff
@@ -5958,12 +5958,13 @@ export declare namespace Tag {
    * @category models
    */
   export interface ProhibitedType {
-    _op?: "propety name _op is forbidden"
-    _tag?: "propety name _tag is forbidden"
-    of?: "propety name of is forbidden"
-    context?: "propety name context is forbidden"
-    key?: "propety name key is forbidden"
-    stack?: "propety name stack is forbidden"
+    _op?: `propety "_op" is forbidden`
```
Comment (leighman) [inline packages/effect/src/Effect.ts:5961]: `property` typo?

Comment (fubhy) [reply packages/effect/src/Effect.ts:5961]: Nah man, that's definitely how it's spelled!

Comment (mikearnaldi) [reply packages/effect/src/Effect.ts:5961]: lol

Comment (mikearnaldi) [reply packages/effect/src/Effect.ts:5961]: https://github.com/Effect-TS/effect/pull/3345

**Pull request merged**